File: LeetCode Daily Challenge/1323. Maximum 69 Number_17 Aug 2025/1323. Maximum 69 Number/C++/main.cpp
class Solution {
    public:
        int maximum69Number (int num) {
            int cur = 0;
            int numC = num;
            int indFirstSix = INT_MIN;
            while(numC!=0) {
                if(numC%10 == 6){
                    indFirstSix = cur;
                }
                numC /= 10;
                cur++;
            }
    
            return indFirstSix == -1 ? num : num + 3 * (int)pow(10, indFirstSix);
        }
    };

File: LeetCode Daily Challenge/1323. Maximum 69 Number_17 Aug 2025/1323. Maximum 69 Number/Go/main.go
package main

import (
	"fmt"
	"math"
)

func maximum69Number(num int) int {
	cur := 0
	numC := num
	indFirstSix := -1

	for numC != 0 {
		if numC%10 == 6 {
			indFirstSix = cur
		}
		numC /= 10
		cur++
	}

	if indFirstSix == -1 {
		return num
	}
	return num + 3*int(math.Pow(10, float64(indFirstSix)))
}

func main() {
	// Test cases
	test1 := 9669
	fmt.Printf("Input: %d\n", test1)
	fmt.Printf("Output: %d\n", maximum69Number(test1))
}
